File: abc/abc127/abc127_b.cs
// https://atcoder.jp/contests/abc127/submissions/8379991

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var input = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
            var r = input[0];
            var D = input[1];
            var x = input[2];

            var result = x;
            for (int i = 0; i < 10; i++)
            {
                result = r * result - D;
                Console.WriteLine(result);
            }
        }
    }
}

File: abc/abc127/abc127_b.fs
// https://atcoder.jp/contests/abc127/submissions/12233589

let [| r; D; x |] = stdin.ReadLine().Split() |> Array.map int

let rec solver t i =
    if i = 10 then ()
    else
        let temp = r * t - D
        temp |> printfn "%d"
        solver temp (i + 1)

solver x 0
